facade/publicendpoint_test.go

```diff
@@ -289,3 +289,91 @@ func (ft *FacadeIntegrationTest) Test_PublicEndpoint_PortEnable(c *C) {
 
 	fmt.Println(" ##### Test_PublicEndpoint_PortEnable: PASSED")
 }
+
+func (ft *FacadeIntegrationTest) Test_PublicEndpoint_VHostAdd(c *C) {
+	fmt.Println(" ##### Test_PublicEndpoint_VHostAdd: STARTED")
+
+	// Add a service so we can test our public endpoint.
+	svcA := service.Service{
+		ID:           "validate-service-tenant-A",
+		Name:         "TestFacade_validateServiceTenantA",
+		DeploymentID: "deployment-id",
+		PoolID:       "pool-id",
+		Launch:       "auto",
+		DesiredState: int(service.SVCStop),
+		Endpoints: []service.ServiceEndpoint{
+			service.ServiceEndpoint{
+				Application: "zproxy",
+				Name:        "zproxy",
+				PortNumber:  8080,
+				Protocol:    "tcp",
+				Purpose:     "export",
+				PortList: []servicedefinition.Port{
+					servicedefinition.Port{
+						PortAddr: ":22222",
+						Enabled:  true,
+						UseTLS:   true,
+						Protocol: "https",
+					},
+				},
+				VHostList: []servicedefinition.VHost{
+					servicedefinition.VHost{
+						Name:    "zproxy",
+						Enabled: true,
+					},
+				},
+			},
+		},
+	}
+	// Add a service so we can test our public endpoint.
+	svcB := service.Service{
+		ID:           "validate-service-tenant-B",
+		Name:         "TestFacade_validateServiceTenantB",
+		DeploymentID: "deployment-id",
+		PoolID:       "pool-id",
+		Launch:       "auto",
+		DesiredState: int(service.SVCStop),
+		Endpoints: []service.ServiceEndpoint{
+			service.ServiceEndpoint{
+				Application: "service2",
+				Name:        "service2",
+				PortNumber:  9090,
+				Protocol:    "tcp",
+				Purpose:     "export",
+			},
+		},
+	}
+	c.Assert(ft.Facade.AddService(ft.CTX, svcA), IsNil)
+	c.Assert(ft.Facade.AddService(ft.CTX, svcB), IsNil)
+
+	// Mock call expectations:
+	ft.zzk.On("CheckRunningPublicEndpoint", registry.PublicEndpointKey(":22222-1"), svcA.ID).Return(nil)
+	ft.zzk.On("CheckRunningPublicEndpoint", registry.PublicEndpointKey("zproxy-0"), svcA.ID).Return(nil)
+	ft.zzk.On("CheckRunningPublicEndpoint", registry.PublicEndpointKey("zproxy2-0"), svcA.ID).Return(nil)
+
+	// Add a vhost to an invalid service.
+	_, err := ft.Facade.AddPublicEndpointVHost(ft.CTX, "invalid", "zproxy", "zproxy", true, true)
```

Comment:
You don't have to change this, but for my $0.02, I prefer to see each of these calls to `AddPublicEndpointVHost()` as separate test cases where the name of the test method clarifies what is being tested.